api-js/src/loaders/domains/load-domain-connections-by-user-id.js

```diff
@@ -0,0 +1,164 @@
+const { aql } = require('arangojs')
+const { fromGlobalId, toGlobalId } = require('graphql-relay')
+
+const domainLoaderConnectionsByUserId = (
+  query,
+  userId,
+  cleanseInput,
+) => async ({ after, before, first, last }) => {
+  let afterTemplate = aql``
+  let beforeTemplate = aql``
+
+  const userDBId = `users/${userId}`
+
+  if (typeof after !== 'undefined') {
+    const { id: afterId } = fromGlobalId(cleanseInput(after))
+    afterTemplate = aql`FILTER TO_NUMBER(domain._key) > TO_NUMBER(${afterId})`
+  }
+
+  if (typeof before !== 'undefined') {
+    const { id: beforeId } = fromGlobalId(cleanseInput(before))
+    beforeTemplate = aql`FILTER TO_NUMBER(domain._key) < TO_NUMBER(${beforeId})`
+  }
+
+  let limitTemplate = aql``
+  if (typeof first !== 'undefined' && typeof last === 'undefined') {
+    limitTemplate = aql`SORT domain._key ASC LIMIT TO_NUMBER(${first + 1})`
+  } else if (typeof first === 'undefined' && typeof last !== 'undefined') {
+    limitTemplate = aql`SORT domain._key DESC LIMIT TO_NUMBER(${last + 1})`
+  } else if (typeof first !== 'undefined' && typeof last !== 'undefined') {
+    console.warn(
+      `User: ${userId} tried to have first and last set in domain connection query`,
+    )
+    throw new Error(
+      'Error, unable to have first, and last set at the same time.',
+    )
+  }
+
+  let affiliatedOrgsCursor
```

Comment:
So I have come up with this new way of querying that reduces the chances of a db, cursor failure and that is to do it all in one step
```
dkimScansCursor = await query`
LET dkimIds = (FOR v, e IN 1 ANY ${domainId} domainsDKIM RETURN e._to)
FOR dkimScan IN dkim
    FILTER dkimScan._id IN dkimIds
    ${afterTemplate}
    ${beforeTemplate}
    ${limitTemplate}
    RETURN dkimScan
`
```

In your case it would look something like
```
LET userAffiliations = (FOR v, e IN 1..1 ANY ${userDBId} affiliations RETURN e._from)
FOR userAffiliation IN userAffilitations
    LET orgClaims = (FOR v, e IN 1..1 ANY userAffiliation claims RETURN e._to)
    FOR domain IN domains
      FILTER orgClaims[** FILTER CURRENT == domain._id]
      ${afterTemplate}
      ${beforeTemplate}
      ${limitTemplate}
      RETURN domain
```